tasks/extract-binaries-from-image/README.md

```diff
@@ -5,13 +5,20 @@ Tekton task that extracts binaries to be released on github.com from an image.
 The path to the directory inside the provided workspace where the binaries were
 saved is provided as a result.
 
+The binaries must be stored at the same `image_binaries_path` for each component
+when multiple components with binaries are to be extracted.
+
 ## Parameters
 
 | Name | Description | Optional | Default value |
 |------|-------------|----------|---------------|
 | image_binaries_path | Path inside the image where the binaries to extract are stored | Yes | /releases |
 | snapshotPath | Path to the JSON string of the mapped Snapshot spec in the data workspace | No | |
 | subdirectory | Subdirectory inside the workspace to be used for storing the binaries | Yes | "" |
+| desired_component_list | List of components inside Snapshot to extract binaries. Includes all by default | Yes | [] |
```

Comment:
I'm trying to understand the use case here. Why would you have multiple components in the snapshot, but only wanted to extract binaries from some of them?

Reply:
Also, how is this gonna be provided? If we want such thing, shouldn't it rather be provided via the data file as we do for all other parameters?

Reply:
Our broad goal is to enable this task to extract binaries that we later ship to RH developer portal.

Some application's Snapshots (RHTAS, for example) have 3 components with CLI binaries that need shipped to developer portal and 2 components that exist to serve the binaries from an openshift instance. These latter two components do not need shipped to the developer portal. 

Making this more granular would help avoid unnecessary churning through images that don't have binaries.

The RPA will contain a `components` key with a list of components to include. My initial thought was the pipeline under development [here](https://github.com/konflux-ci/release-service-catalog/pull/435): could pass the components list after the `collect-data` task populates the data with something like `$(tasks.collect-data.results.data.components_list)`

But now that I look for examples, I'm thinking maybe accessing a nested value like that wont work? The ReleasePlanAdmission data is available from the workspace, right? I don't need to pass it as a parameter?

If so, is that your suggestion here?

Reply:
The ReleasePlanAdmission is stored as a json in the workspace by collect-data. You don't need to pass it as a parameter, but the relative path to it should be passed as we use subdirectories to avoid collisions. Also, the data file is what you usually want instead of the RPA, as the data file contains the merged result of the keys

Reply:
In your example of 5 components, 3 with binaries, do we need to ship the other 2 or are those just ignored? If the latter, we could just not map them; then they wouldn't be in the mapped snapshot

Reply:
So IIUC:

In pipeline, pass this to the task:
```
- name: dataPath
          value: "$(tasks.collect-data.results.data)"
```
Then in the task,  accept it as a param and access it like this:

`DATA_FILE="$(workspaces.data.path)/$(params.dataPath)"`

Reply:
As for the components, the 2 remaining in that scenario I believe are still expected to be delivered to the registry in another pipeline. So I'd imagine they'll need a second RPA for those 2 components.

pkhander has a design doc up for review showing the RPA with a content gateway key. One of the proposed values for this field is a list of components. That's how I imagined the desired components would originate.

You idea of having it as `binary: yes` in the `data:mapping:components` section could also work.

Reply:
> So IIUC:
> 
> In pipeline, pass this to the task:
> 
> ```
> - name: dataPath
>           value: "$(tasks.collect-data.results.data)"
> ```
> 
> Then in the task, accept it as a param and access it like this:
> 
> `DATA_FILE="$(workspaces.data.path)/$(params.dataPath)"`

Yes, exactly